### DISCUSSION TITLE
Welcome to nginx-s3-gateway Discussions!

### DISCUSSION THREAD

**Discussion by dekobon** [nginxinc/nginx-s3-gateway#17] [General] [answered]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by digitalvintage** [comment, marked as answer]
Hi! Nice app, I like to use it as Minio S3 Browser, but I experienced problem with cyrillic object names in S3. I just got Error 404 when accessing directory with cyrillic name or trying to download a file. Is it a bug or just some configuration needed (and I just had not enough attention)?

**Reply by dekobon** [reply]
Hi there, could you please provide some example strings that are failing? We have tests that use Japanese, so in theory unicode object names should work.

**Reply by digitalvintage** [reply]
> Hi there, could you please provide some example strings that are failing? We have tests that use Japanese, so in theory unicode object names should work.

![изображение](https://user-images.githubusercontent.com/57669494/155011614-f18572bc-83d1-4e9a-b9cd-27bdb4986aa5.png)
And the link is "http://retro.digitalvintage.ru:88/%D0%9E%D0%BF%D0%B5%D1%80%D0%B0%D1%86%D0%B8%D0%BE%D0%BD%D0%BD%D1%8B%D0%B5%20%D1%81%D0%B8%D1%81%D1%82%D0%B5%D0%BC%D1%8B/" or "http://retro.digitalvintage.ru:88/Операционные системы/"
But none of cyrillic links on screenshot aren't working, leading to 404 page.

**Reply by dekobon** [reply]
Is the backend AWS S3 or another S3 compatible object store?

**Reply by digitalvintage** [reply]
> Is the backend AWS S3 or another S3 compatible object store?

It's Minio

**Reply by dekobon** [reply]
In the `default.conf.template`, under the `@s3Listing` location, can you comment out or remove these lines:
```
        proxy_intercept_errors on;

        # Comment out this line to receive the error messages returned by S3
        error_page 400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 420 422 423 424 426 428 429 431 444 449 450 451 500 501 502 503 504 505 506 507 508 509 510 511 =404 @error404;
```
Then rerun the failing request. This time an error message should be provided in the response. If you can get that error message, can you please post it here?

**Reply by digitalvintage** [reply]
> 
> Then rerun the failing request. This time an error message should be provided in the response. If you can get that error message, can you please post it here?

Mmmm...it said
"500 Internal Server Error
junk"

![изображение](https://user-images.githubusercontent.com/57669494/155018403-f5dbeeb1-2679-486a-9273-f0c57bd81d50.png)

**Reply by dekobon** [reply]
Are you able to access the objects directly from Minio on the same server that nginx is running using something like the aws cli or [`s3cmd`](https://s3tools.org/s3cmd)?

**Reply by digitalvintage** [reply]
> Are you able to access the objects directly from Minio on the same server that nginx is running using something like the aws cli or [`s3cmd`](https://s3tools.org/s3cmd)?
![изображение](https://user-images.githubusercontent.com/57669494/155021893-cc80412f-9005-48d5-b594-b7e8a258fd12.png)

![изображение](https://user-images.githubusercontent.com/57669494/155021777-027fc424-f6c1-44e1-9516-0d0d095a4b1a.png)
It works...

**Reply by dekobon** [reply]
I just attempted to reproduce the issue on my system running the gateway, linux and Minio dockerized version `minio/minio:RELEASE.2021-02-19T04-38-02Z`. I created a directory named `системы` and placed a file called `system.txt` in it and I was able to access it via the gateway.

Is Minio running on Linux with ext4 as the file system?

Also, can you check the Minio logs to see what may be causing the 500 error? Typically 500 errors mean there is a problem with the service. If the request was bad, it would return a 4xx error.

**Reply by dekobon** [reply]
Also, is this running in the Windows Linux subsystem by chance?

**Reply by digitalvintage** [reply]
> I just attempted to reproduce the issue on my system running the gateway, linux and Minio dockerized version `minio/minio:RELEASE.2021-02-19T04-38-02Z`. I created a directory named `системы` and placed a file called `system.txt` in it and I was able to access it via the gateway.
> 
> Is Minio running on Linux with ext4 as the file system?
> 
> Also, can you check the Minio logs to see what may be causing the 500 error? Typically 500 errors mean there is a problem with the service. If the request was bad, it would return a 4xx error.

My setup:
Minio dockerized (official container) version 2022-02-12T00:51:25Z
OS CentOS 7.9 / Linux 3.10.0-1160.53.1.el7.x86_64 (host)
Filesystem - ext4

Unfortunately no errors in Minio logs and traces except this one:
API: ListObjectsV1
Time: 18:43:41:0
DeploymentID: 5b43ff3b-c6e8-4a7d-ab9a-2a3a6033cc8a
RequestID: 16D592C6BF62B873
RemoteHost: 172.18.0.1
Host: s3.digitalvintage.ru:443
Error: Object name invalid: digitalvintage/������������%20�������/ (cmd.ObjectNameInvalid)
cmd/object-api-input-checks.go:75:cmd.checkListObjsArgs()
cmd/object-api-common.go:235:cmd.listObjects()
cmd/fs-v1.go:1270:cmd.(*FSObjects).ListObjects()
cmd/bucket-listobjects-handlers.go:347:cmd.objectAPIHandlers.ListObjectsV1Handler()
net/http/server.go:2047:http.HandlerFunc.ServeHTTP()

Windows is only on this workstation and it's used for browser and ssh access to server.

**Reply by dekobon** [reply]
I'm now able to reproduce the error. It happens with both AWS S3 and Minio. If you try to access an object under a unicode prefix/directory, it will return without problems. However, if you try to list the contents of a prefix/directory, it will return the error reported above. The string `junk` is returned from this line: https://github.com/nginxinc/nginx-s3-gateway/blob/master/common/etc/nginx/include/s3gateway.js#L291

**Reply by dekobon** [reply]
I just created this issue to track the problem: https://github.com/nginxinc/nginx-s3-gateway/issues/25

Let's move further discussion to that issue.

**Reply by digitalvintage** [reply]
> I just created this issue to track the problem: #25
> 
> Let's move further discussion to that issue.

Thank you!
